pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -244,6 +244,9 @@ func (p *Provisioner) reconcileCluster(cluster *v3.Cluster, create bool) (*v3.Cl
 			logrus.Infof("Create done, Updating cluster [%s]", cluster.Name)
 			apiEndpoint, serviceAccountToken, caCert, err = p.driverUpdate(cluster, *spec)
 		}
+	} else if spec.RancherKubernetesEngineConfig.Restore.Restore {
+		logrus.Infof("Restoring cluster [%s] from backup", cluster.Name)
+		apiEndpoint, serviceAccountToken, caCert, err = p.driverRestore(cluster, *spec)
```

Comment:
is restore one time operation? if yes, when update goes through successfully, need to set make `Restore` false somewhere here https://github.com/rancher/rancher/blob/486b190eb52ae8b5b7ca1cec840fab39b44202c8/pkg/controllers/management/clusterprovisioner/provisioner.go#L289